app/models/application_data.rb

```diff
@@ -0,0 +1,265 @@
+class ApplicationData
+  def initialize(data)
+    @data = data || {}
```

Comment:
Again personal preference, but I'd make this a default and keeping the assignments as simple as possible:
```ruby
def initialize(data = {})
  @data = data
end
```